### PR TITLE
test: getTemplateFuncMap unit tests part 3

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Ensures that the following `getTemplateFuncMap` function keys are thinly wrapped go functions with unit test coverage:

```
		"GetProbes",
		"GetSecurityRules",
		"GetUniqueNameSuffix",
		"GetVNETAddressPrefixes",
		"GetVNETSubnets",
		"GetDataDisks",
		"HasBootstrap",
```

Also removed the entirely unnecessary `HasBootstrapPublicIP` and `IsHostedBootstrap` static `return false` template functions.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/engine/engine_test.go [hunk 1]
```diff
@@ -672,28 +697,6 @@ func TestGenerateKubeConfig(t *testing.T) {
 	}
 }
 
-func TestGetDataDisks(t *testing.T) {
```
Comment (jackfrancis) [inline pkg/engine/engine_test.go]: busted this was not an actual test :)

File: pkg/engine/template_generator.go [hunk 2]
```diff
@@ -355,13 +355,7 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 			return getDataDisks(profile)
 		},
 		"HasBootstrap": func() bool {
-			return cs.Properties.OrchestratorProfile.DcosConfig != nil && cs.Properties.OrchestratorProfile.DcosConfig.BootstrapProfile != nil
-		},
-		"HasBootstrapPublicIP": func() bool {
-			return false
```
Comment (CecileRobertMichon) [inline pkg/engine/template_generator.go]: good one

File: parts/dcos/bstrap/bootstrapparams.t [hunk 3]
```diff
@@ -12,22 +12,6 @@
       "type": "securestring"
       },
     {{end}}
-{{if IsHostedBootstrap}}
```
Comment (CecileRobertMichon) [inline parts/dcos/bstrap/bootstrapparams.t]: I agree this change is pretty straight forward but I still think we should hold ourselves to not changing any legacy DCOS code without having an e2e test.

Comment (jackfrancis) [reply parts/dcos/bstrap/bootstrapparams.t]: I think this is just unacceptable cruft we should get rid of in order to streamline the code surface area.

**Pull request merged**